flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveSchedulerTest.java

```diff
@@ -1080,10 +1053,10 @@ private Iterable<RootExceptionHistoryEntry> runExceptionHistoryTests(
                             declarativeSlotPool,
                             createSlotOffersForResourceRequirements(
                                     ResourceCounter.withResource(
-                                            ResourceProfile.UNKNOWN, numAvailableSlots)),
+                                            ResourceProfile.UNKNOWN, PARALLELISM)),
                             taskManagerGateway);
                 });
-        listener.waitForRunning();
+        taskManagerGateway.waitForSubmissions(4, TestingUtils.infiniteDuration());
```

Comment:
can we add a comment here about why waiting for all subtasks here is necessary due to checkpointing?

Reply:
good point; will do